synapse/storage/background_updates.py

```diff
@@ -273,9 +410,14 @@ async def _do_background_update(self, desired_duration_ms: float) -> int:
         if items_per_ms is not None:
             batch_size = int(desired_duration_ms * items_per_ms)
             # Clamp the batch size so that we always make progress
-            batch_size = max(batch_size, self.MINIMUM_BACKGROUND_BATCH_SIZE)
+            batch_size = max(
+                batch_size,
+                await self._min_batch_size(update_name, self._database_name),
+            )
         else:
-            batch_size = self.DEFAULT_BACKGROUND_BATCH_SIZE
+            batch_size = await self._default_batch_size(
+                update_name, self._database_name
+            )
```

Comment:
In part yes. It's also because you never know what we might want to condition the returned value on, e.g. maybe in the future EMS will want to customise the batch size depending on how much RDS IO credits the instance has at that specific point in time, etc.